lib/handlers/revoke-handler.ts

```diff
@@ -0,0 +1,341 @@
+import * as auth from 'basic-auth';
+import {
+  InvalidArgumentError,
+  InvalidClientError,
+  InvalidRequestError,
+  InvalidTokenError,
+  OAuthError,
+  ServerError,
+} from '../errors';
+import { Client, Model } from '../interfaces';
+import { Request } from '../request';
+import { Response } from '../response';
+import { oneSuccess } from '../utils/fn';
+import * as is from '../validator/is';
+
+export class RevokeHandler {
+  model: Model;
+  constructor(options: any = {}) {
```

Comment:
It seems that the type of options is actually `{model: Model}`, as it is required, and there is no other property of `options` being picked here. As such then it is odd that the code allows to not provide `options`, as the default value clearly will fail.